integrations/nuclei_analyzer/app.py

```diff
@@ -0,0 +1,240 @@
+            "success": False,
+            "error": "An unexpected error occurred",
+            "details": str(e),
+            "scan_status": "ERROR",
+        }
+
+
+def async_nuclei_scan(task_id: str, url: str, template_dirs: list = None):
+    try:
+        _, report = run_nuclei_command(url, template_dirs)
+
+        task_store[task_id].update(
+            {
+                "status": "completed",
+                "report": report,
+                "completed_at": datetime.utcnow().isoformat(),
+            }
+        )
+        logger.info(f"Task {task_id} reports stored successfully")
+    except Exception as e:
+        task_store[task_id].update(
+            {
+                "status": "error",
+                "error": str(e),
+                "completed_at": datetime.utcnow().isoformat(),
+            }
+        )
+
+
+@app.route("/run-nuclei", methods=["GET"])
+def get_nuclei_scan_status():
+    task_id = request.args.get("key")
+    if not task_id:
+        return jsonify({"success": False, "error": "Missing task key parameter"}), 400
+
+    if task_id not in task_store:
+        return jsonify({"success": False, "error": "Invalid task key"}), 404
+
+    task_info = task_store[task_id]
+
+    if task_info["status"] == "running":
+        return (
+            jsonify({"status": "running", "started_at": task_info["started_at"]}),
+            200,
+        )
+
+    return (
+        jsonify(
+            {
+                "status": task_info["status"],
+                "report": task_info.get("report"),
+                "error": task_info.get("error"),
+                "started_at": task_info["started_at"],
+                "completed_at": task_info["completed_at"],
+            }
+        ),
+        200,
+    )
+
+
+@app.route("/run-nuclei", methods=["POST"])
+def start_nuclei_scan():
+    try:
+        data = request.get_json()
+
+        if not data or "observable" not in data:
+            return (
+                jsonify(
+                    {
+                        "success": False,
+                        "error": "Invalid request, 'observable' field is required",
+                    }
+                ),
+                400,
+            )
+
+        obs = data["observable"]
+        template_dirs = data.get("template_dirs", [])
+
+        if not isinstance(template_dirs, list):
+            return (
+                jsonify(
+                    {
+                        "success": False,
+                        "error": "Invalid request, 'template_dirs' must be a list",
+                    }
+                ),
+                400,
+            )
+
+        task_id = str(uuid.uuid4())
+        task_store[task_id] = {
+            "status": "running",
+            "started_at": datetime.utcnow().isoformat(),
+        }
+
+        thread = Thread(target=async_nuclei_scan, args=(task_id, obs, template_dirs))
+        thread.start()
+
+        return jsonify({"status": "accepted", "key": task_id}), 200
```

Comment:
as mentioned before, this cannot work like that. A single API call from the client must be enough to handle all the  stuff and give back the results of the analysis. Plus, everything must be keep synchronous.

Reply:
so, you should write a polling routine here to wait for the task to be completed and get the results back